lib/utils.bash

```diff
@@ -576,13 +576,15 @@ with_plugin_env() {
 plugin_executables() {
   local plugin_name=$1
   local full_version=$2
-  for bin_path in $(list_plugin_exec_paths "$plugin_name" "$full_version"); do
+  local bin_path all_bin_paths
+  all_bin_paths="$(list_plugin_exec_paths "$plugin_name" "$full_version")"
+  while read -r bin_path; do
```

Comment:
The same thing could be done here as well. All we need to do is read `all_bin_paths` line by line since paths are delimited by newlines.